### DISCUSSION TITLE
Why I see ??? under location in summary reporter

### DISCUSSION THREAD

**Discussion by Partho** [bloomberg/memray#619] [Q&A] [answered]
Based on the numbers attached in the screenshot, I assume we're not memory constrained on any of this stuff, but I see ??? for the largest allocation. Any idea why that's happening? We are curious because its a large allocation, it will be quite helpful if you point out what leads to the loss of allocation info. Also, it will be great if you can mention a way we retrieve the details of function and location that got obfuscated by ???

![Screenshot 2024-05-31 at 10 29 31 PM](https://github.com/bloomberg/memray/assets/2034301/adb4d07a-62fb-4a57-bc0b-6135de0e4d2a)

**Comment by godlygeek** [comment, marked as answer]
That means that the allocation occurred at a point when there were no Python frames on the thread's stack. If you enable [native tracking](https://bloomberg.github.io/memray/run.html#native-tracking) you can get capture a stack that includes native frames as well as Python frames, which would help you tell where that allocation occurred as long as debug symbols are available for whatever library it occurred in (probably libc or libpython, since that seems to have happened early in the creation of a new thread).
